### ISSUE TITLE
android.os.DeadObjectException

### ISSUE THREAD

**Issue by trebouillon** [adjust/android_sdk#266]
This crash occurs while startup of the app and the SDK tries to get the app info. This happens on almost 99% on devices with Marshmallow and mostly Samsung as manufacturer.

I am using "**com.adjust.sdk:adjust-android-criteo:4.7.0**".

0 java.lang.RuntimeException: Package manager has died

1 at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:198)

2 at com.adjust.sdk.DeviceInfo.getAppVersion(DeviceInfo.java:107)

3 at com.adjust.sdk.DeviceInfo.<init>(DeviceInfo.java:64)

4 at com.adjust.sdk.ActivityHandler.initInternal(ActivityHandler.java:477)

5 at com.adjust.sdk.ActivityHandler.access$000(ActivityHandler.java:31)

6 at com.adjust.sdk.ActivityHandler$1.run(ActivityHandler.java:126)

7 at android.os.Handler.handleCallback(Handler.java:739)

8 at android.os.Handler.dispatchMessage(Handler.java:95)

9 at android.os.Looper.loop(Looper.java:158)

10 at android.os.HandlerThread.run(HandlerThread.java:61)

11 Caused by: android.os.DeadObjectException

12 at android.os.BinderProxy.transactNative(Native Method)

13 at android.os.BinderProxy.transact(Binder.java:503)

14 at android.content.pm.IPackageManager$Stub$Proxy.getPackageInfo(IPackageManager.java:2721)

15 at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:193)

16 ... 9 more

17 android.os.DeadObjectException

18 at android.os.BinderProxy.transactNative(Native Method)

19 at android.os.BinderProxy.transact(Binder.java:503)

20 at android.content.pm.IPackageManager$Stub$Proxy.getPackageInfo(IPackageManager.java:2721)

21 at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:193)

22 at com.adjust.sdk.DeviceInfo.getAppVersion(DeviceInfo.java:107)

23 at com.adjust.sdk.DeviceInfo.<init>(DeviceInfo.java:64)

24 at com.adjust.sdk.ActivityHandler.initInternal(ActivityHandler.java:477)

25 at com.adjust.sdk.ActivityHandler.access$000(ActivityHandler.java:31)

26 at com.adjust.sdk.ActivityHandler$1.run(ActivityHandler.java:126)

27 at android.os.Handler.handleCallback(Handler.java:739)

28 at android.os.Handler.dispatchMessage(Handler.java:95)

29 at android.os.Looper.loop(Looper.java:158)

30 at android.os.HandlerThread.run(HandlerThread.java:61)

**Comment by Obaied** [comment]
Hey @trebouillon.
Can you try to update to the latest version of Criteo (v4.11.0)?
Here's the doc: https://github.com/adjust/android_sdk/blob/master/doc/english/criteo_plugin.md

Also, where do you store your app's version? Our SDK retrieves the app's version the `versionName` parameter inside your app's `build.gradle`. Please refer to this page for a code example: https://developer.android.com/studio/publish/versioning.html

**Comment by trebouillon** [comment]
I didn't try the newest version so far. But I can check this with my next major release. And the version name is set in my build gradle.

**Comment by Obaied** [comment]
The exception is happening because the PackageManager couldn't retrieve the app version. My guess is that one of two things is the problem:
- app version is not getting read properly for some reason
- the context object that is getting supplied to our SDK (from `AdjustConfig`) is somehow not being able to retrieve the package manager (thus creating the *DeadObjectException* that `package manager has died` since it can't find).

Is it possible to provide us with:
- Your `android` code block inside your app's `build.gradle` which includes the `versionName` attribute. It should look like this: 
```
android {
    compileSdkVersion 25
    buildToolsVersion "25.0.2"

    defaultConfig {
        minSdkVersion 9
        targetSdkVersion 25
        versionCode 1
        versionName getVersionName()
    }
}
```

- Also, the code snippet where you initialise our `AdjustConfig` object and call `Adjust.onCreate(config)`. Please include the name of the function and class you're calling it from, as well.

Thanks for reporting this issue.

**Comment by trebouillon** [comment]
Sorry for my late answer. This is how its look like in my app:

```
android {
    compileSdkVersion 25
    buildToolsVersion "25.0.2"

    defaultConfig {
        versionCode 123
        versionName '4.5.6'
        minSdkVersion 16
        targetSdkVersion 25
        …
    }
}
```

And the initialization of Adjust:

```
    final AdjustConfig adjustConfig = new AdjustConfig(context, „abcdef12345“, AdjustConfig.ENVIRONMENT_PRODUCTION);
    adjustConfig.setLogLevel(LogLevel.ASSERT);
    Adjust.onCreate(adjustConfig);
```

In my case the context is the application context. When the `onCreate()` method of the application class is called Adjust will be initialized.

**Comment by Obaied** [comment]
Can you provide us with the version of adjust SDK are you using? you mentioned which criteo-adjust version, but not the SDK version.

**Comment by trebouillon** [comment]
It is the same. I only include the Criteo Gradle dependency and this includes the same version of Adjust SDK, or am I wrong?

**Comment by Obaied** [comment]
I believe so. Is it possible to try the following...
- Remove the criteo gradle dependency
- Add those two dependencies:
```
compile 'com.adjust.sdk:adjust-android:4.11.0'
compile 'com.adjust.sdk:adjust-android-criteo:4.11.0'
```

Let me know how it goes :)

**Comment by Obaied** [comment]
Worst case scenario if that doesn't work, we are releasing a new version that has proper exception handling for retrieving app versions so you won't have this crash anymore. We're already working on it.

**Comment by trebouillon** [comment]
I will give it a try with the next release. This might take a while :)

**Comment by uerceg** [comment]
This should now be resolved with latest Android SDK v4.11.1 release (https://github.com/adjust/android_sdk/issues/266).

Feel free to reopen the issue if the problem still happens to occur with newest SDK version.

Cheers
